src/diagnostics.rs

```diff
@@ -165,9 +194,17 @@ pub fn report_error<'tcx, 'mir>(
         &if let Some(title) = title { format!("{}: {}", title, msg) } else { msg.clone() },
         msg,
         helps,
-        &ecx.generate_stacktrace(),
+        &stacktrace,
     );
 
+    // Include a note like `std` does for short backtraces, but since we are opt-out not opt-in, we
+    // do not include a note when backtraces are off.
+    if ecx.machine.backtrace_style == BacktraceStyle::Short {
+        ecx.tcx.sess.diagnostic().note_without_error(
+            "Some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace.",
```

Comment:
```suggestion
            "some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace",
```
following the usual style of Rust diagnostics

Reply:
Maybe someone should also complain about `std`'s inconsistency :wink:

Reply:
std is not emitting rustc diagnostics. ;)